common/testing/testlogger/testlogger.go

```diff
@@ -324,10 +340,7 @@ func (tl *TestLogger) Debug(msg string, tags ...tag.Tag) {
 	if tl.state.mu.closed {
 		return
 	}
-	if tl.tags != nil {
-		tags = append(tags, tl.tags...)
-	}
-	tl.wrapped.Debug(msg, append(tags, tl.tags...)...)
+	tl.wrapped.Debug(msg, tl.mergeTags(tags)...)
```

Comment:
any idea what this change does to the performance of (in particular) Debug()?  Because this merging happens whether Debug is enabled or not, and it's often called in VERY high-cardinality places.

OTOH, it _is_ a test logger, so maybe it's not critical